File: src/main/java/kitchenpos/products/infrastructure/ProductRepositoryImpl.java
package kitchenpos.products.infrastructure;

import java.util.List;
import java.util.Optional;
import java.util.UUID;
import kitchenpos.products.domain.Product;
import kitchenpos.products.domain.ProductRepository;
import org.springframework.stereotype.Repository;

@Repository
public class ProductRepositoryImpl implements ProductRepository {

    private final JPAProductRepository repository;

    public ProductRepositoryImpl(JPAProductRepository repository) {
        this.repository = repository;
    }

    @Override
    public Product save(Product product) {
        return repository.save(product);
    }

    @Override
    public Optional<Product> findById(UUID productId) {
        return repository.findById(productId);
    }

    @Override
    public List<Product> findAllByIdIn(List<UUID> ids) {
        return null;
    }

    @Override
    public List<Product> findAll() {
        return repository.findAll();
    }
}

File: src/main/java/kitchenpos/products/domain/ProductRepositoryImpl.kt
package kitchenpos.products.domain

import org.springframework.stereotype.Repository
import java.util.*

@Repository
class ProductRepositoryImpl(
    private val productJpaRepository: ProductJpaRepository
) : ProductRepository {
    override fun save(product: Product): Product {
        return productJpaRepository.save(product.toRecord()).toProduct()
    }

    override fun findById(id: UUID?): Optional<Product> {
        TODO("Not yet implemented")
    }

    override fun findAll(): MutableList<Product> {
        TODO("Not yet implemented")
    }

    override fun findAllByIdIn(ids: MutableList<UUID>?): MutableList<Product> {
        TODO("Not yet implemented")
    }
}

fun Product.toRecord(): ProductRecord {
    return ProductRecord(
        id,
        getName(),
        getPrice(),
    )
}

fun ProductRecord.toProduct(): Product {
    return Product(
        id = id,
        name = ProductName.create(name),
        price = ProductPrice.create(price)
    )
}
